offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/products/ProductsDbConnector.groovy

```diff
@@ -273,23 +273,59 @@ class ProductsDbConnector implements ArchiveProductDataSource, CreateProductData
    * @throws ProductExistsException if the product already exists in the data source
    */
   @Override
-  void store(Product product) throws DatabaseQueryException, ProductExistsException {
+  ProductId store(Product product) throws DatabaseQueryException, ProductExistsException {
```

Comment:
To remove the dependency on the assumption that the provided product exactly matches the stored product (some fields are not stored or modified somehow e.g. sample-tracking date, offer date), and thereby removing the need to query the database in a second call to fetch the product information for forwarding it to the use case output, I would suggest

* returning the `Product` directly
```suggestion
  Product store(Product product) throws DatabaseQueryException, ProductExistsException {
```
* or documenting the assumption and providing only the `uniqueId` of the newly created ProductId (product type does not change and productId generation could be viewed as an overhead).

Reply:
In any case, please provide matching JavaDoc comments to the interface method (adding a `@return` here)

Reply:
huh.. why is it not here. I updated the java doc 🤔

Reply:
I did that for the interface. Since when do we have methods descriptions for API methods?